switchboard/manager.py

```diff
@@ -83,12 +85,37 @@ def __getitem__(self, key):
         """
         return SwitchProxy(self, super(SwitchManager, self).__getitem__(key))
 
+    def with_result_cache(func):
+        """
+        Decorator specifically for is_active.  If self.result_cache is set to a {}
+        the is_active results will be cached for each set of params.
+        """
+        def inner(self, *args, **kwargs):
+            dic = self.result_cache
+            cache_key = None
+            if dic is not None:
+                cache_key = (args, tuple(kwargs.items()))
+                try:
+                    result = dic.get(cache_key)
+                except TypeError as e:  # not hashable
```

Comment:
For my own edification, what's the significance of TypeError being not hashable?